src/Column/RangeColumn.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Pgsql\Column;
+
+use InvalidArgumentException;
+use Yiisoft\Db\Exception\NotSupportedException;
+use Yiisoft\Db\Expression\ExpressionInterface;
+use Yiisoft\Db\Pgsql\Constant\PgsqlColumnType;
+use Yiisoft\Db\Pgsql\Expression\DateRangeValue;
+use Yiisoft\Db\Pgsql\Expression\Int4RangeValue;
+use Yiisoft\Db\Pgsql\Expression\Int8RangeValue;
+use Yiisoft\Db\Pgsql\Expression\NumRangeValue;
+use Yiisoft\Db\Pgsql\Expression\TsRangeValue;
+use Yiisoft\Db\Pgsql\Expression\TsTzRangeValue;
+use Yiisoft\Db\Schema\Column\AbstractColumn;
+
+use function gettype;
+use function is_array;
+use function is_string;
+use function sprintf;
+
+final class RangeColumn extends AbstractColumn
+{
+    protected const DEFAULT_TYPE = PgsqlColumnType::RANGE;
+
+    public function dbTypecast(mixed $value): mixed
+    {
+        if ($value === null || $value instanceof ExpressionInterface) {
+            return $value;
+        }
+
+        if (!is_array($value)) {
+            $this->throwWrongTypeException(gettype($value));
+        }
+
+        if (array_keys($value) !== [0, 1]) {
+            throw new InvalidArgumentException(
+                'Value of range should be an array with two elements: lower and upper bounds.',
+            );
+        }
+
+        $dbType = $this->getDbType();
+        [$lower, $upper] = $value;
+
+        $column = match ($dbType) {
+            'int4range' => ColumnBuilder::integer(),
+            'int8range' => ColumnBuilder::bigint(),
+            'numrange' => ColumnBuilder::decimal(),
+            'tsrange' => ColumnBuilder::datetime(),
+            'tstzrange' => ColumnBuilder::datetimeWithTimezone(),
+            'daterange' => ColumnBuilder::date(),
+            default => $this->throwNotSupportedDbType($dbType),
+        };
+
+        return '['
+            . $this->dbTypecastBound($column->dbTypecast($lower))
+            . ','
+            . $this->dbTypecastBound($column->dbTypecast($upper))
+            . ')';
+    }
+
+    public function phpTypecast(mixed $value): mixed
+    {
+        if ($value === null || $value === 'empty') {
+            return null;
+        }
+
+        if (!is_string($value)) {
+            $this->throwWrongTypeException(gettype($value));
+        }
+
```

Comment:
```suggestion
```

We expect `phpTypecast()` to only receive the value that the database returns, which in this case is `null` or a `string`. To avoid extra checks.